core/src/main/java/org/apache/carbondata/core/metadata/blocklet/BlockletInfo.java

```diff
@@ -205,7 +205,7 @@ public void setNumberOfPages(int numberOfPages) {
     output.writeLong(dimensionOffset);
     output.writeLong(measureOffsets);
     int dsize = dimensionChunkOffsets != null ? dimensionChunkOffsets.size() : 0;
-    output.writeShort(dsize);
+    output.writeInt(dsize);
```

Comment:
Why you have changed this.
This is for number of dimension columns, how it will change for varchar data type ??

Reply:
OK~